Rework OSDK intros + Golang examples (#28372)
openshift/openshift-docs: modules/osdk-golang-multi-group-apis.adoc

// Module included in the following assemblies:
//
// * operators/operator_sdk/osdk-golang-tutorial.adoc

[id="osdk-golang-multi-group-apis_{context}"]
= About multi-group APIs

> Before creating an API and controller, consider whether your Operator requires multiple API groups. This tutorial covers the default case of a single group API, but to change the layout of your project to support multi-group APIs, you can run the following command:

Reviewer: ```suggestion
Before you create an API and controller, consider whether your Operator requires multiple API groups. This tutorial covers the default case of a single group API, but to change the layout of your project to support multi-group APIs, you can run the following command:
```

[source,terminal]
----
$ operator-sdk edit --multigroup=true
----

> This command updates the `PROJECT` file, which should look like the following:

Reviewer: ```suggestion
This command updates the `PROJECT` file, which should look like the following example:
```

[source,yaml]
----
domain: example.com
layout: go.kubebuilder.io/v3
multigroup: true
...
----

> For multi-group projects, the API Go type files are created under `apis/<group>/<version>/`, and the controllers are created under `controllers/<group>/`. The Dockerfile is then updated accordingly.

Reviewer: ```suggestion
For multi-group projects, the API Go type files are created in the `apis/<group>/<version>/` directory, and the controllers are created in the `controllers/<group>/` directory. The Dockerfile is then updated accordingly.
```

.Additional resource

> * For more details on migrating to multi-group, see the link:https://book.kubebuilder.io/migration/multi-group.html[Kubebuilder documentation].

Reviewer: ```suggestion
* For more details on migrating to a multi-group project, see the link:https://book.kubebuilder.io/migration/multi-group.html[Kubebuilder documentation].
```
